src/main.py

```
@@ -348,114 +356,132 @@ async def knock(
+    requested_ttl = (
+        getattr(body, "ttl", None) if hasattr(body, "ttl") else (body.get("ttl") if body else None)
+    )
+
     effective_ttl = max_ttl
 
     if requested_ttl is not None:
         # Comprehensive TTL validation
         if not isinstance(requested_ttl, int):
-            logging.warning(f"Invalid TTL type '{type(requested_ttl).__name__}' specified by {client_ip}.")
+            logging.warning(
+                f"Invalid TTL type '{type(requested_ttl).__name__}' specified by {client_ip}."
```

Ruff (0.15.13)

[warning] Logging statement uses f-string

(G004)